modules/persistent-storage-snapshots-creating-deleting.adoc

```diff
@@ -0,0 +1,124 @@
+
+Depending on the PV type, the `Create Snapshot` operation might go through several phases, which are reflected by the `VolumeSnapshot` status:
+
+. Create the new `VolumeSnapshot` object.
+. Start the controller. The snapshotted `PersistentVolume` might need to be frozen and the applications paused.
+. Create ("cut") the snapshot. The snapshotted `PersistentVolume` might return to normal operation, but the snapshot itself is not yet ready (status=`True`, type=`Pending`).
+. Create the new `VolumeSnapshotData` object, representing the actual snapshot.
+. The snapshot is complete and ready to use (status=`True`, type=`Ready`).
+
+[IMPORTANT]
+====
+It is the user's responsibility to ensure data consistency (stop the Pod or application, flush caches, freeze the file system, and so on).
+====
+
+[NOTE]
+====
+In case of error, the `VolumeSnapshot` status is appended with an `Error` condition.
+====
+
+To display the `VolumeSnapshot` status:
+
+----
+$ oc get volumesnapshot -o yaml
+----
+The status is displayed, as shown in the following example:
+
+[source,yaml]
+----
+apiVersion: volumesnapshot.external-storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  clusterName: ""
+  creationTimestamp: 2017-09-19T13:58:28Z
+  generation: 0
+  labels:
+    Timestamp: "1505829508178510973"
+  name: snapshot-demo
+  namespace: default
+  resourceVersion: "780"
+  selfLink: /apis/volumesnapshot.external-storage.k8s.io/v1/namespaces/default/volumesnapshots/snapshot-demo
+  uid: 9cc5da57-9d42-11e7-9b25-90b11c132b3f
+spec:
+  persistentVolumeClaimName: ebs-pvc
+  snapshotDataName: k8s-volume-snapshot-9cc8813e-9d42-11e7-8bed-90b11c132b3f
+status:
+  conditions:
+  - lastTransitionTime: null
+    message: Snapshot created successfully
+    reason: ""
+    status: "True"
+    type: Ready
+  creationTimestamp: null
+----
+== Restore snapshot
+
+A PVC is used to restore a snapshot. But first, the administrator must create a `StorageClass` to restore a `PersistentVolume` from an existing `VolumeSnapshot`.
+
+. Create a `StorageClass`:
++
+[source,yaml]
+----
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: snapshot-promoter
+provisioner: volumesnapshot.external-storage.k8s.io/snapshot-promoter
+----
+
+. Create a PVC:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: snapshot-pv-provisioning-demo
+  annotations:
+    snapshot.alpha.kubernetes.io/snapshot: snapshot-demo <1>
+spec:
+  storageClassName: snapshot-promoter <2>
+----
+<1> The name of the `VolumeSnapshot` to be restored.
+<2> Created by the administrator for restoring `VolumeSnapshots`.
++
+A new `PersistentVolume` is created and bound to the `PersistentVolumeClaim`.
+The process might take several minutes depending on the PV type.
+
+== Delete snapshot
+
+To delete a `snapshot-demo`:
+
+----
+$ oc delete volumesnapshot/snapshot-demo
```

Comment:
This is fine as is, but we could also make it generic:

> $ oc delete volumesnapshot/<snapshot-name>

Reply:
Went with your suggestion.